nomos-blend/network/src/core/with_core/behaviour/tests/utils.rs

```diff
@@ -45,41 +48,63 @@ impl IntervalProviderBuilder {
     }
 }
 
-#[derive(Default)]
+/// Generates `count` nodes with randomly generated identities and empty
+/// addresses.
+pub fn new_nodes_with_empty_address(
+    count: usize,
+) -> (impl Iterator<Item = Keypair>, Vec<Node<PeerId>>) {
+    let mut identities: Vec<Keypair> = repeat_with(Keypair::generate_ed25519).take(count).collect();
+    identities.sort_by_key(|id| PeerId::from(id.public()));
+    let nodes = identities
+        .iter()
+        .map(|identity| Node {
+            id: identity.public().into(),
+            address: Multiaddr::empty(),
+            public_key: Ed25519PrivateKey::generate().public_key(),
```

Comment:
Should we not use the public key of the identity instead?

Reply:
Now I got what you mean. Yes. Let's use the same key: 1bcdbe50a9adb93dbf8ff9a87e7ae752d64428e9